corehq/apps/translations/app_translations/utils.py

```diff
@@ -257,3 +246,54 @@ def update(self, rows):
         a function like django.contrib.messages.error, and the second is a string.
         '''
         raise NotImplementedError()
+
+
+# TODO: (2019-04-26) Drop support for legacy names after all translations could possibly have been uploaded
+#
+def get_legacy_name_map(app, exclude_module=None, exclude_form=None):
+    """
+    Maps legacy module and form sheet names to current names
+    """
+    name_map = {}
+    for module in app.get_modules():
+        if exclude_module is not None and exclude_module(module):
+            continue
+        legacy_name = get_module_legacy_sheet_name(module)
+        name_map[legacy_name] = get_module_sheet_name(module)
+
+        for form in module.get_forms():
+            if exclude_form is not None and exclude_form(form):
+                continue
+            legacy_name = get_form_legacy_sheet_name(form)
+            name_map[legacy_name] = get_form_sheet_name(form)
+
+    return name_map
+
+def get_module_legacy_sheet_name(module):
+    """
+    Returns "menuN" where N was its (1-based) index.
+
+    Breaks if module is moved. Use get_module_sheet_name() instead.
+    """
+    return "menu{}".format(module.get_app().get_module_index(module.unique_id) + 1)
+
+
+def get_form_legacy_sheet_name(form):
+    """
+    Returns "menuM_formN" where N was its (1-based) index.
+
+    Breaks if form or module is moved. Use get_form_sheet_name() instead.
+    """
+    module = form.get_module()
+    return "_".join([
+        get_module_sheet_name(module),
+        "form{}".format(module.get_form_index(form.unique_id) + 1)
+    ])
+
+
+def is_legacy_form_sheet(identifier):
+    return ('module' in identifier or 'menu' in identifier) and 'form' in identifier
+
+
+def is_legacy_module_sheet(identifier):
```

Comment:
Won't this also return True for non-legacy module sheet names (`'menu' in 'menu:UUID:slug' == True`)? Most of the time it looks like you call `is_module_sheet` before `is_legacy_module_sheet`, but this usage looks like it might be problematic: https://github.com/dimagi/commcare-hq/pull/24151/files#diff-b55921fb78674519bb781aa227828617R196